tensorflow/python/ops/math_ops.py

```diff
@@ -822,8 +822,10 @@ def real(input, name=None):
     if input.dtype.is_complex:
       real_dtype = input.dtype.real_dtype
       return gen_math_ops.real(input, Tout=real_dtype, name=name)
-    else:
+    elif tf.debugging.is_numeric_tensor(input):
```

Comment:
We can't use `tf.` from within here, since it's not imported (and can't be, since that would be a circular dependency).  

You can import the function itself from [`check_ops.py`](https://github.com/tensorflow/tensorflow/blob/395ee9f672bb1417447d135c55e9a676773afa15/tensorflow/python/ops/check_ops.py#L1954) where it actually lives.  Or, we could add a property to dtype itself for `is_numeric`, which is probably the most general solution.  Those properties are defined in [`dtypes.cc`](https://github.com/tensorflow/tensorflow/blob/395ee9f672bb1417447d135c55e9a676773afa15/tensorflow/python/framework/dtypes.cc#L130), and we already have a `kNumberTypes` set in [`types.h`](https://github.com/tensorflow/tensorflow/blob/395ee9f672bb1417447d135c55e9a676773afa15/tensorflow/core/framework/types.h#L261) that can be leveraged (see `DataTypeIsComplex(...)` in that file as an example).

Reply:
I just implemented these changes. Does it look fine now? Also, is there any specific ordering to functions in dtypes.cc or types.h to keep it consistent with other files? For the time being, I placed the code for is_numeric and IsNumericDataType right above the code for is_complex and IsComplexDataType.

Reply:
Thanks, looks good now.  No special ordering, as long as its somewhat consistent.

Will run through further testing now.

Reply:
Hi. I'm still waiting on Py+CPP Test Suite - Ubuntu CPU, Python 3.9 and Code Check - Changed Files workflows to be run and approved. It seems odd that it's taking this long. Any ideas why?

Reply:
Yes, sorry, it broke our API compatibility tests.  It shouldn't be a big issue because you only added a read-only attribute, but I'll need to go in and update the API checks, and do some more testing to make sure save/load models doesn't break.

Reply:
Okay, thanks. Let me know if I need to change anything else.